lib/truffle/digest.rb

```diff
@@ -136,17 +136,23 @@ def inspect
   class Class
     include Instance
 
-    def self.digest(message)
-      digest = new
+    def self.digest(message, *parameters)
+      digest = new(*parameters)
       digest.update message
       digest.digest
     end
 
-    def self.hexdigest(message)
-      digest = new
+    def self.hexdigest(*parameters, message)
+      digest = new(*parameters)
       digest.update message
       digest.hexdigest
     end
```

Comment:
`Digest.hexdigest` seems defined as `Digest.hexencode(Digest::Class.digest(*args))` in MRI:
https://github.com/ruby/ruby/blob/37c2cd3fa47c709570e22ec4dac723ca211f423a/ext/digest/digest.c#L478-L490

And then https://github.com/oracle/truffleruby/blob/59b78fdbebd53a5cf9d0a4306d26714ffab23e81/lib/mri/openssl/digest.rb#L34-L36 is what, very confusingly, swaps the arguments.

I'll add a commit to do that.

Reply:
Yes this definitely was a brainbender 😅 also had a fun time with how `Digest::Class` and `Digest::Instance` are used depending on using the stdlib Digest or the OpenSSL algorithm modules:

```ruby
OpenSSL::Digest::SHA256.ancestors
=> [OpenSSL::Digest::SHA256, OpenSSL::Digest, Digest::Class, Digest::Instance, Object, Kernel, BasicObject]

Digest::SHA256.ancestors
=> [Digest::SHA256, Digest::Base, Digest::Class, Digest::Instance, Object, Kernel, BasicObject]

Digest::SHA256.ancestors - OpenSSL::Digest::SHA256.ancestors
=> [Digest::SHA256, Digest::Base]
```